Tests/FluentAssertions.Specs/Xml/XDocumentAssertionSpecs.cs

```diff
@@ -1009,5 +1027,55 @@ public void When_asserting_a_document_has_a_null_element_it_should_fail()
         }
 
         #endregion
+
+        #region xsd
+
+        [Fact]
+        public void Xsd_validation_for_valid_xml_should_work()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<PurchaseOrder xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance'>
+                    <ShipTo>
+                        <name>Shepard</name>
+                        <street>Street</street>
+                    </ShipTo>
+                    <BillTo>
+                        <name>Billy</name>
+                        <street>Street</street>
+                    </BillTo>
+                </PurchaseOrder>");
+
+            // Act
+            Action act = () => document.Should().BeValidByXsd(XsdSample);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void Xsd_validation_for_invalid_xml_should_throw()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<PurchaseOrder xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance'>
+                    <BillTo>
+                        <name>Billy</name>
+                        <street>Street</street>
+                    </BillTo>
+                    <ShipTo>
+                        <name>Shepard</name>
+                        <street>Street</street>
+                    </ShipTo>
+                </PurchaseOrder>");
+
+            // Act
+            Action act = () => document.Should().BeValidByXsd(XsdSample, " because we want to test the failure message");
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("Expected XML document to match XSD because we want to test the failure message*");
```

Comment:
I would at least assert that a fragment of the expected error message from the XSD validation is there.

Reply:
On one hand I agree.
On the other hand I want to note, that the validation message will be language specific. So the test will depend on the locale settings of the executing host.
Maybe the test should be executed within fixed thread locale.

Reply:
Yep, I didn't assert internal xsd error, because I have it in Russian

Reply:
You can use the `[CulturedFact]` attribute to force